frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -147,9 +147,18 @@ local VirtualKeyboard = InputContainer:new{
     key_padding = Screen:scaleBySize(6),
 }
 
+-- international keyboard
 local keyboard_layout = require("ui/data/keyboardlayouts/std")
+-- polish keyboard
+local keyboard_layout_pl = require("ui/data/keyboardlayouts/pl_keyboard")
+
 function VirtualKeyboard:init()
-    self.KEYS = keyboard_layout
+    local lang = G_reader_settings:readSetting("language")
+    if lang == "pl" then  --polish keyboard
+        self.KEYS = keyboard_layout_pl
+    else -- international
+        self.KEYS = keyboard_layout
+    end
```

Comment:
I suggest we create a language to keyboard module mapping table here so only one keyboard layout is loaded at runtime. Something like the following:

```lua
local lang_to_keyboard_layout = {
    pl = "pl_keyboard",
}

local keyboard_layout = lang_to_keyboard_layout[lang] or "std"
self.KEYS = require("ui/data/keyboardlayouts/" .. keyboard_layout)
```